packages/formation-react/src/helpers/link-utils.js

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+
+/**
+ * Return JSX for an accessibly hidden icon (tested in AccessibleLink)
+ * @param {string} className - Fontawesome icon group & class, e.g.
+ *  "fas fa-download"
+ * @return {ReactElement} - icon element
+ */
+export const createIcon = className => (
+  <i
+    role="img"
+    aria-hidden="true"
+    className={`vads-u-padding-right--1 ${className}`}
+  />
+);
+
+/**
+ * *** DownloadLink utils ***
+ * (tested in DownloadLink utils)
+ */
+/**
+ * Non-optimal method to find the file name & extension. It splits off the final
+ * forward slash (`/`) and processes that portion of the content.
+ * @param {string} url
+ * @return {string} last portion of a url; after the last forward slash
+ */
+export const getFileNameAndExt = url => url.split('/').pop();
+
+// Regex to extract a file extension of a url path
+// https://www.example.com/test.jpg -> jpg
+// https://www.foo.bar.com/my.folder/doc.pdf?version=3.4 -> pdf
+// https://www.test.com/file.txt#anchor-link -> txt
+const fileExtRegex = /(?:\.([^.?#]+))?(?:[?#].*?)?$/;
+
+/**
+ * Extract file extension from a url
+ * @param {string} url - path to downloadable file
+ * @return {string} - Known file extension. If not found, this function returns
+ * an empty string
+ */
+export const getFileExtensionFromUrl = fileName => {
+  const ext = fileExtRegex.exec(fileName);
+  return ext && ext[1] ? ext[1] : '';
+};
+
+// Regex used to replace digits & separators to extract out the file size unit
+// "1.5mb" -> "mb"
+const fileSizeUnitRegex = /[0-9., ]/g;
+
+// List of known file size units
+const FILE_SIZES = [
+  // Decimal (SI) / Binary (IEC)
+  ['B', 'Bytes', 'B', 'Bytes'],
+  ['KB', 'Kilobytes', 'KiB', 'Kibibytes'],
+  ['MB', 'Megabytes', 'MiB', 'Mebibytes'],
+  ['GB', 'Gigabytes', 'GiB', 'Gibibytes'],
+];
+
+/**
+ * Extract file size data from a mixed value of number plus size unit, e.g.
+ * "1.5mb"
+ * @param {string|number} size - file size, e.g. "1.5mb" or 1024 (bytes)
+ * @return {ReactElement} rendered size with unit abbr & full unit in title
+ */
+export const renderFileSize = (size = null, si = true, decimalPlaces = 1) => {
```

Comment:
I don't know how I missed adding `si` and `decimalPlaces` to the JSDoc! 

I tend to avoid using external packages for small things because there is usually so much code-bloat... but maybe [Sindre's pretty-bytes](https://github.com/sindresorhus/pretty-bytes) package would be okay since it has no dependencies?

Reply:
I think if it's a well-supported dependency with a really narrow focus and won't add much to the end bundle size, I think adding an external package is fine. It looks like [pretty-bytes](https://www.npmjs.com/package/pretty-bytes) is quite popular and small, so I'd be :+1: to adding that if it solves the problem.